qm2/scf.py

```diff
@@ -0,0 +1,113 @@
+import numpy as np
+import psi4
+
+np.set_printoptions(suppress=True, precision=4)
+
+
+def build_geom(mol_string):
+    mol = psi4.geometry(mol_string)
+    
+    # Build a molecule
+    mol.update_geometry()
+    mol.print_out()
+    return mol 
+
+e_conv = 1.e-6
+d_conv = 1.e-6
+nel = 5
+damp_value = 0.20
+damp_start = 5
+
+# Build a basis
+bas = psi4.core.BasisSet.build(mol,target="aug-cc-pvdz")
+bas.print_out()
+
+# Build a MintsHelper
+mints = psi4.core.MintsHelper(bas)
+nbf=mints.nbf()
+
+if (nbf > 100):
+   raise Exception("More than 100 basis functions!")
+
+print(mints.ao_potential())
+
+V = np.array(mints.ao_potential())
+T = np.array(mints.ao_kinetic())
+
+#Core Hamiltonian
+H = T + V
+
+S = np.array(mints.ao_overlap())
+g = np.array(mints.ao_eri())
+
+#print(S.shape)
+#print(g.shape)
+
+
+A = mints.ao_overlap()
+A.power(-0.5, 1.e-14)
+A = np.array(A)
+
+# print(A @ S @ A)
+
+def diag(F,A):
+    Fp = A.T @ F @ A
+    eps, Cp = np.linalg.eigh(Fp)
+    C = A @ Cp
+    return eps, C
+
+eps, C = diag(H,A)
+Cocc = C[:, :nel]
+D = Cocc @ Cocc.T
+
+
+
+E_old = 0.0
+F_old = H
+
+def fock_build(D, damp=False):
+    J = np.einsum("pqrs,rs->pq",g,D)
+    K = np.einsum("prqs,rs->pq", g,D)
+    F_new = H + 2.0 * J - K
+    if damp:
+        F = damp_value * F_old + (1.0-damp_value) * F_new
```

Comment:
How does this function know what `damp_value` is, should it be an argument? It could also just be hard-coded since right now it is not able to be changed by the user at run time, (it has a static constant value)